s3torchconnector/tst/unit/lightning/test_s3_lightning_checkpoint.py

```diff
@@ -6,62 +6,105 @@
+    return s3_lightning_checkpoint
+
+
+@pytest.fixture()
+def async_lightning_checkpoint(lightning_checkpoint) -> AsyncCheckpointIO:
+    return AsyncCheckpointIO(lightning_checkpoint)
+
+
+@hypothesis.settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
 @given(python_primitives, byteorders)
-def test_lightning_checkpointing_saves_python_primitives(data, byteorder):
-    _test_save(data, byteorder)
+def test_lightning_checkpointing_saves_python_primitives(client, lightning_checkpoint, data, byteorder):
+    _test_save(client, lightning_checkpoint, data, byteorder)
+
+
+@hypothesis.settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+@given(byteorders)
+def test_lightning_checkpointing_saves_tensor(client, lightning_checkpoint, byteorder):
+    tensor = torch.rand(2, 4)
+    _test_save(client, lightning_checkpoint, tensor, byteorder, equal=torch.equal)
 
 
+@hypothesis.settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
 @given(byteorders)
-def test_lightning_checkpointing_saves_tensor(byteorder):
-    tensor = torch.tensor([[0.1, 1.2], [2.2, 3.1], [4.9, 5.2]])
-    _test_save(tensor, byteorder, equal=torch.equal)
+def test_async_lightning_checkpointing_saves_tensor(client, lightning_checkpoint, byteorder):
+    tensor = torch.rand(2, 4)
+    _test_save(client, lightning_checkpoint, tensor, byteorder, equal=torch.equal)
 
 
+@hypothesis.settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
 @given(byteorders)
-def test_lightning_checkpointing_saves_untyped_storage(byteorder):
+def test_lightning_checkpointing_saves_untyped_storage(client, lightning_checkpoint, byteorder):
     storage = torch.UntypedStorage([1, 2, 3])
     _test_save(
+        client,
+        lightning_checkpoint,
         storage,
         byteorder,
         equal=lambda a, b: list(a) == list(b),
     )
 
 
+@hypothesis.settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
 @given(python_primitives, byteorders)
-def test_lightning_checkpointing_loads_python_primitives(data, byteorder):
-    _test_load(data, byteorder)
+def test_lightning_checkpointing_loads_python_primitives(client, lightning_checkpoint, data, byteorder):
+    _test_load(client, lightning_checkpoint, data, byteorder)
+
+
+@hypothesis.settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+@given(byteorders)
+def test_lightning_checkpointing_loads_tensor(client, lightning_checkpoint, byteorder):
+    tensor = torch.rand(2, 4)
+    _test_load(client, lightning_checkpoint, tensor, byteorder, equal=torch.equal)
 
 
+@hypothesis.settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
 @given(byteorders)
-def test_lightning_checkpointing_loads_tensor(byteorder):
-    tensor = torch.tensor([[0.1, 1.2], [2.2, 3.1], [4.9, 5.2]])
-    _test_load(tensor, byteorder, equal=torch.equal)
+def test_lightning_checkpointing_async_loads_tensor(client, async_lightning_checkpoint, byteorder):
```

Comment:
nit: the name is inconsistent with `test_async_lightning_checkpointing_saves_tensor`
We may not need that test at all, as `AsyncCheckpointIO` doesn't provide async method for loading path.